src/librustc/ty/mod.rs

```diff
@@ -965,6 +965,21 @@ impl GenericParamDefKind {
             GenericParamDefKind::Const => "constant",
         }
     }
+
+    /// Is this a const parameter?
+    pub fn is_const(&self) -> bool {
+        if let GenericParamDefKind::Const = self { true } else { false }
+    }
+
+    /// Is this a lifetime parameter?
+    pub fn is_lifetime(&self) -> bool {
+        if let GenericParamDefKind::Lifetime = self { true } else { false }
+    }
+
+    /// Is this a type parameter?
+    pub fn is_type(&self) -> bool {
+        if let GenericParamDefKind::Type { .. } = self { true } else { false }
+    }
```

Comment:
Well-intentioned, but can lead to fragile code.
Note that you can use `matches!` nowadays if you want.

Reply:
Fixed in 9b391f1.